src/server/database/votes.ts

```diff
@@ -112,33 +119,78 @@ export default class Votes {
             const { insertId } = row;
 
             // If clip has enough votes or is from super user update it
-            this.updateClipStatus(clipId, isSuper, vote);
+            if (updateClip) {
+                this.updateClipStatus(clipId, isSuper, vote);
+            }
 
             return Promise.resolve(insertId);
         } catch (error) {
             return Promise.reject(error);
         }
     };
 
-    addVoteBatch = async (votes: Vote[]): Promise<number> => {
+    addVoteBatch = async (votes: Vote[]): Promise<BatchUploadResults> => {
         try {
+            const output: BatchUploadResults = new BatchUploadResults();
             // Find Marosijo user_client or create a new one if none exists
             const marosijo_client_id = await this.findMarosijoClientId();
+            const validatedIds = await this.getValidatedClips(votes);
+
+            // Store for output
+            output.foundValidatedClips = validatedIds.length;
+
+            // Filter out clips that have been validated
+            const toAdd = votes.filter((vote) => {
+                return !validatedIds.some(
+                    (validated) => validated == vote.clipId
+                );
+            });
+
+            // Create an array for the VALUES that are to be added to the votes table
+            const values: string[] = Array<string>();
+            toAdd.forEach((vote) => {
+                const unsure = ClipVote.UNSURE == vote.vote;
+                const isValid = unsure
+                    ? null
+                    : vote.vote == ClipVote.VALID
+                    ? true
+                    : false;
+                values.push(
+                    `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`
+                );
+            });
+
+            // Add votes to votes table
+            const res = await this.sql.query(`
+                    INSERT INTO
+                        votes (id, clip_id, client_id, is_valid, is_super, is_unsure)
+                    VALUES
+                        ${values.join(',')}
```

Comment:
INSERT INTO
        votes (id, clip_id, client_id, is_valid, is_super, is_unsure)
    VALUES
        ${values.join(',')}

Just to be clear, it this what happens?

INSERT INTO
        votes (id, clip_id, client_id, is_valid, is_super, is_unsure)
    VALUES
        `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`,
        `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`,
        `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`,
        `(null, ${vote.clipId}, '${marosijo_client_id}', ${isValid}, ${vote.isSuper}, ${unsure} )`,
        ...

?

Reply:
Yes basically the lines above create each value as a string.
And then here we get a long list of values (rows) to add (well we add in batches of 500, so that is the max)

All votes here should be added regardless if they update a clip or not. And I have already filtered out any vote that has a clip_id where the clip has a valid in the is_valid column in the clips table.